File: app/src/main/java/co/wm21/https/presenter/application/BDPStatusPresenter.java
package co.wm21.https.presenter.application;

import co.wm21.https.FHelper.networks.ApiUtil.OnRequestComplete;
import co.wm21.https.FHelper.networks.Models.application.BDPStatusResponse;
import co.wm21.https.FHelper.networks.Models.application.EShopRefComResponse;
import co.wm21.https.presenter.interfaces.aplication.OnBDPStatusView;
import co.wm21.https.presenter.interfaces.aplication.OnEShopRefComView;
import co.wm21.https.serviceapis.InvokeBDPStatusApi;

public class BDPStatusPresenter {
    OnBDPStatusView onView;

    public BDPStatusPresenter(OnBDPStatusView onView) {
        this.onView = onView;
    }

    public void getDataLoad(String memberID) {
        onView.onStartLoading();
        new InvokeBDPStatusApi(memberID, new OnRequestComplete() {
            @Override
            public void onRequestSuccess(Object obj) {
                onView.onStopLoading();
                onView.onBDPStatusDataLoad((BDPStatusResponse) obj);
            }

            @Override
            public void onRequestError(String errMsg) {
                onView.onStopLoading();
                onView.onError(errMsg);
            }
        });
    }
}

File: app/src/main/java/co/wm21/https/presenter/application/BDPStatusPresenter.kt
package co.wm21.https.presenter.application

import co.wm21.https.FHelper.networks.ApiUtil.OnRequestComplete
import co.wm21.https.FHelper.networks.Models.application.BDPStatusResponse
import co.wm21.https.presenter.interfaces.aplication.OnBDPStatusView
import co.wm21.https.serviceapis.InvokeBDPStatusApi

class BDPStatusPresenter(var onView: OnBDPStatusView) {
    fun getDataLoad(memberID: String?) {
        onView.onStartLoading()
        InvokeBDPStatusApi(memberID, object : OnRequestComplete {
            override fun onRequestSuccess(obj: Any) {
                onView.onStopLoading()
                onView.onBDPStatusDataLoad(obj as BDPStatusResponse)
            }

            override fun onRequestError(errMsg: String) {
                onView.onStopLoading()
                onView.onError(errMsg)
            }
        })
    }
}
